### ISSUE TITLE
Support for HalfTensor

### ISSUE THREAD

**Issue by grow-yhq** [zhanghang1989/PyTorch-Encoding#36]
Hi, your work is really helpful to me, it's a incredible project. But when I found that Pytorch-Encoding don't support HalfTensor, it brought me some pain. My model applying synchronous batch normalization with FloatTensor got a satisfied outcome, so I want to enlarge batch size and slim the model size by using HalfTensor and compare this two models. Limited by my knowledge on cuda, I couldn't figure it out myself, so I come here telling my request. I know that you're quite busy,  if this disturb you, I'm sorry.

**Comment by zhanghang1989** [comment]
Thanks @grow-yhq 

1) If you'd like to use FP16, you can add Generate FP16 options https://github.com/zhanghang1989/PyTorch-Encoding/blob/master/encoding/kernel/thc_encoding.cu#L26

2) But what people typically do is using FP16 for Convolution layer and cast the data to FP32 for BN layers.

**Comment by grow-yhq** [comment]
Thanks for your advice! My problem solved using the second way.
